tests/test_bot.py

```diff
@@ -1431,3 +1431,93 @@ def assertion(url, data, *args, **kwargs):
         monkeypatch.setattr(bot.request, 'post', assertion)
 
         assert bot.close()
+
+    @flaky(3, 1)
+    @pytest.mark.timeout(10)
+    def test_copy_message(self, monkeypatch, bot, chat_id, media_message):
+        keyboard = InlineKeyboardMarkup(
+            [[InlineKeyboardButton(text="test", callback_data="test2")]]
+        )
+
+        def post(url, data, timeout):
+            assert data["chat_id"] == chat_id
+            assert data["from_chat_id"] == chat_id
+            assert data["message_id"] == media_message.message_id
+            assert data["caption"] == "<b>Test</b>"
+            assert data["parse_mode"] == ParseMode.HTML
+            assert data["reply_to_message_id"] == media_message.message_id
+            assert data["reply_markup"] == keyboard
+            return data
+
+        monkeypatch.setattr(bot.request, 'post', post)
+        bot.copy_message(
+            chat_id,
+            from_chat_id=chat_id,
+            message_id=media_message.message_id,
+            caption="<b>Test</b>",
+            parse_mode=ParseMode.HTML,
+            reply_to_message_id=media_message.message_id,
+            reply_markup=keyboard,
+        )
+
+    @flaky(3, 1)
+    @pytest.mark.timeout(10)
+    def test_copy_message_without_reply(self, bot, chat_id, media_message):
```

Comment:
what exactly was the problem here to combine the two tests? shouldn't something like

```python
orig_post = bot.requst.post

def post(*args, **kwargs):
    ...
    return orig_post(*args, **kwargs)
```
do the trick? if it's indeed complicated for some reason, just leave it as is, though …

Reply:
The issue is here that it also tries to do the test in post for the temp send message request. I dont know man I hate tests. Im just gonna let it be split

Reply:
ah okay. yeah, np